WordPress/gutenberg
# Move InserterButton from components package to block-editor package

## What?
This PR moves the mobile `InserterButton` from the components package to the block-editor package.

## Why?
There is an ongoing effort to remove block-editor package dependencies in native files from the components package in order to eliminate cyclic dependencies:

* https://github.com/WordPress/gutenberg/issues/52692
* https://github.com/WordPress/gutenberg/issues/56072
* https://github.com/WordPress/gutenberg/issues/27751

## How?
Moves code and code references to `InserterButton` from the components package to the block-editor package.

## Testing Instructions
1. Create a post and use the Block Inserter button (bottom left corner of screen)
2. Button should function as normal compared to production
